src/iter/fold.rs

```diff
@@ -114,4 +118,8 @@ impl<'r, C, ID, F, T> Folder<T> for FoldFolder<'r, C, ID, F>
     fn complete(self) -> C::Result {
         self.base.consume(self.item).complete()
     }
+
+    fn full(&self) -> bool {
+        self.base.full()
+    }
```

Comment:
Fair point, and those that don't get full are logically infinite given infinite input.  I guess we should make that required rather than implicit.